core/validation/pre_transformation_validate.py

```diff
@@ -0,0 +1,168 @@
+    :param place_names: A tuple of valid place names.
+    :param fund_types: A tuple of valid fund types.
+    :param schema: The schema containing authorisation checks.
+    :raises ValidationError: If authorisation validation fails.
+    """
+    type_to_authorised_values = {
+        "Place Name": place_names,
+        "Fund Type": fund_types,
+    }
+    authorisation_checks = list()
+    for sheet, column, row, expected_values, type in schema["authorisation_checks"]:
+        expected_values = type_to_authorised_values[type]
+
+        authorisation_checks.append(
+            PreTransformationCheck(sheet=sheet, column=column, row=row, expected_values=expected_values, type=type)
+        )
+
+    failures = check_values(workbook, authorisation_checks, "unauthorised_submission_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
+
+
+def wrong_input_validation(workbook: dict[str, pd.DataFrame], schema: dict):
+    """Performs wrong input validation based on the provided schema.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the submission.
+    :param schema: The schema containing wrong input checks.
+    :raises ValidationError: If wrong input validation fails.
+    """
+    wrong_input_checks = schema["wrong_input_checks"]
+
+    failures = check_values(workbook, wrong_input_checks, "wrong_input_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
+
+
+def conflicting_input_validation(workbook: dict[str, pd.DataFrame], schema: dict):
+    """Performs conflicting input validation based on the provided schema.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the submission.
+    :param schema: The schema containing conflicting input checks.
+    :raises ValidationError: If conflicting input validation fails.
+    """
+    fund_type_acronym_to_full_name = {"TD": "Town_Deal", "HS": "Future_High_Street_Fund"}
+    conflicting_input_checks = list()
+    for sheet, column, row, expected_values, type in schema["conflicting_input_checks"]:
+        # TODO: if extra conflicting checks introduced, make this this logic dependent on 'type'
+        col, row = expected_values
+        fund_type_of_entered_place = PLACE_TO_FUND_TYPE.get(str(workbook[sheet].iloc[col][row]).strip())
+        expected_values = tuple(map(fund_type_acronym_to_full_name.get, fund_type_of_entered_place))
+
+        conflicting_input_checks.append(
+            PreTransformationCheck(sheet=sheet, column=column, row=row, expected_values=expected_values, type=type)
+        )
+
+    failures = check_values(workbook, conflicting_input_checks, "wrong_input_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
```

Comment:
I think this function still needs some more thought.

It's currently still quite unclear as to what it does. It also doesn't look very reusable because it's tightly coupled to the `PLACE_TO_FUND_TYPE` mapping.

This is possibly down to the nature of the check we're trying to do. If it's really specific like this then I'm not sure we should support it here.

Reply:
It's to deal with a edge case that turned up in QA prior to the launch of Submit. 

If a user enters a place_name is that one of our valid place_names from our mapping, and they select a valid fund_type, but the place_name does not match the fund_type then this will be flagged, but to get the place_name's fund type we need to do a lookup on the place's fund_type, and then make sure the fund_type is not the acronym but how it appears on the spreadsheet. Unless we envision this mapping incorporating other funds then it won't be reusable; or, at least, the function will have to come with if-statement on 'type'. 

It's actually very difficult to hit this error as the dropdown on the spreadsheet won't let a user select HS for a place only associated with Town_Deal. You have to do some very specific copy-and-pasting to get "Future_High_Street_Fund" in the fund_type box when you've selected Newark as a place. 

Other option is ```towns_fund_round_4_specific_pre_validation.py``` with this function in.

Reply:
I have managed to refactor the function to be fund agnostic. It now has no explicit tie to TF. Quite what we may use this function for in the future is hard to envision without knowing what ingesting another fund actually looks like, but the pre-transformation failures no longer make any explicit reference to TF-only variables.

Reply:
I think this is an improved implementation - nice one